src/main/scala/org/broadinstitute/dsde/agora/server/ga4gh/ModelSupport.scala

```diff
@@ -0,0 +1,120 @@
+package org.broadinstitute.dsde.agora.server.ga4gh
+
+import org.broadinstitute.dsde.agora.server.AgoraConfig
+import org.broadinstitute.dsde.agora.server.ga4gh.Models.{Metadata, Tool, ToolClass, ToolDescriptor, ToolDescriptorType, ToolId, ToolVersion}
+import org.broadinstitute.dsde.agora.server.model.{AgoraEntity, AgoraEntityType, MethodDefinition}
+import wdl4s.WdlNamespaceWithWorkflow
+
+import scala.util.Success
+
+/**
+  * Support class to handle apply methods with some degree of business logic
+  */
+object ModelSupport {
+
+  final val organization = "The Broad Institute of Harvard and MIT"
+  final val verifiedSource = ""
+  final val version = "1.0.0"
+  final val apiVersion = version
+  final val country = "USA"
+  final val friendlyName = "FireCloud"
+
+  def toolIdFromEntity(entity:AgoraEntity): ToolId = {
+    assert(entity.namespace.nonEmpty, "cannot create a ToolId if entity namespace is empty")
+    assert(entity.name.nonEmpty, "cannot create a ToolId if entity name is empty")
+    new ToolId(entity.namespace.get, entity.name.get)
+  }
+
+  def toolIdFromMethod(method:MethodDefinition): ToolId = {
+    assert(method.namespace.nonEmpty, "cannot create a ToolId if method definition namespace is empty")
+    assert(method.name.nonEmpty, "cannot create a ToolId if method definition name is empty")
+    new ToolId(method.namespace.get, method.name.get)
+  }
+
+  def toolClassFromEntityType(entityType: Option[AgoraEntityType.EntityType]): ToolClass = {
+    assert(entityType.nonEmpty, "cannot create a ToolClass if entity type is empty")
+    val str = entityType.get.toString
+    new ToolClass(str, str, "")
+  }
+
+  def toolFromEntities(entities:Seq[AgoraEntity]): Tool = {
+    val representative = entities.last
+    val versions = entities.toList map (x => ModelSupport.toolVersionFromEntity(x))
+    val latestVersion = versions.last
+    val id = ToolId(representative).toString
+    val url = AgoraConfig.GA4GH.toolUrl(id, latestVersion.id, latestVersion.`descriptor-type`.last)
+    val author = findAuthorsInWdl(representative.payload)
+    new Tool(
+      url=url,
+      id=id,
+      organization=organization,
+      toolname=latestVersion.name,
+      toolclass=ToolClass(representative),
+      description=representative.synopsis.getOrElse(""),
+      author=author,
+      `meta-version` = latestVersion.`meta-version`,
+      contains=List.empty[String],
```

Comment:
Since you're already parsing the WDL ... I had considered populating the `contains` field with the names of the tasks contained in the WDL. What do you think?

Reply:
I like that ... will dig into the spec and see what we can get.